storage/interface.go

```diff
@@ -158,13 +158,13 @@ func (f QueryableFunc) Querier(ctx context.Context, mint, maxt int64) (Querier,
 // Operations on the Appender interface are not goroutine-safe.
 type Appender interface {
 	// Append adds a sample pair for the given series.
-	// An optional reference number can be provided to accelerate calls.
-	// A reference number is returned which can be used to add further
-	// samples in the same or later transactions.
-	// Returned reference numbers are ephemeral and may be rejected in calls
-	// to Append() at any point. Adding the sample via Append() returns a new
-	// reference number.
-	// If the reference is 0 it must not be used for caching.
+	// An optional reference number representing series can be provided
+	// instead of label pair to accelerate append calls. Use default value
```

Comment:
It is not "instead of", but "alongside".  If the reference is valid then it will be used and the labels ignored, and if not the labels will be used.

Reply:
Yup 👍🏽  will change.